src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -67,6 +67,7 @@ impl Step for Std {
             target,
             self.override_build_kind.unwrap_or(builder.kind),
         );
+        cargo.arg("--features=std/panic-unwind");
```

Comment:
I think the root cause why this is necessary is that the sysroot crate enables the `panic_unwind` feature of std, rather than the panic-unwind feature. The `panic_unwind` feature exists because the `panic-unwind` feature enables the `panic_unwind` dependency using `panic-unwind = ["panic_unwind"]` rather than `panic-unwind = ["dep:panic_unwind"]`.

Reply:
Try this diff:

```diff
diff --git a/library/std/Cargo.toml b/library/std/Cargo.toml
index fe601855cc1..a025a4ea04e 100644
--- a/library/std/Cargo.toml
+++ b/library/std/Cargo.toml
@@ -90,8 +90,8 @@ backtrace = [
     'miniz_oxide/rustc-dep-of-std',
 ]
 
-panic-unwind = ["panic_unwind"]
-profiler = ["profiler_builtins"]
+panic-unwind = ["dep:panic_unwind"]
+profiler = ["dep:profiler_builtins"]
 compiler-builtins-c = ["alloc/compiler-builtins-c"]
 compiler-builtins-mem = ["alloc/compiler-builtins-mem"]
 compiler-builtins-no-asm = ["alloc/compiler-builtins-no-asm"]
diff --git a/library/sysroot/Cargo.toml b/library/sysroot/Cargo.toml
index 7165c3e48af..b05f61fedcd 100644
--- a/library/sysroot/Cargo.toml
+++ b/library/sysroot/Cargo.toml
@@ -20,7 +20,7 @@ compiler-builtins-no-f16-f128 = ["std/compiler-builtins-no-f16-f128"]
 compiler-builtins-mangled-names = ["std/compiler-builtins-mangled-names"]
 llvm-libunwind = ["std/llvm-libunwind"]
 system-llvm-libunwind = ["std/system-llvm-libunwind"]
-panic-unwind = ["std/panic_unwind"]
+panic-unwind = ["std/panic-unwind"]
 panic_immediate_abort = ["std/panic_immediate_abort"]
 optimize_for_size = ["std/optimize_for_size"]
 profiler = ["std/profiler"]
```

Reply:
Thanks, added at https://github.com/rust-lang/rust/pull/128359/commits/7446e7fd542a56daf80b757d0ac6f082b1ab7d4f

Reply:
It may work without the `cargo.arg("--features=std/panic-unwind");` now.